otter/models/cass.py

```diff
@@ -805,7 +800,12 @@ def view_state(self, consistency=None):
                           NoSuchScalingGroupError(self.tenant_id, self.uuid),
                           self.log)
 
-        return d.addCallback(_unmarshal_state)
+        def check_deleting(group):
```

Comment:
This function is also in `view_manifest`. Do you think you can extract it out and use it in both places?